tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SNS/PublishIntegration.cs

```diff
@@ -0,0 +1,77 @@
+// <copyright file="PublishIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SNS
+{
+    /// <summary>
+    /// AWSSDK.SNS Publish CallTarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.SimpleNotificationService",
+        TypeName = "Amazon.SimpleNotificationService.AmazonSimpleNotificationServiceClient",
+        MethodName = "Publish",
+        ReturnTypeName = "Amazon.SimpleNotificationService.Model.PublishResponse",
+        ParameterTypeNames = new[] { "Amazon.SimpleNotificationService.Model.PublishRequest" },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsSnsCommon.IntegrationName)]
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class PublishIntegration
+    {
+        private const string Operation = "Publish";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TPublishRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the SNS operation</param>
+        /// <returns>CallTarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TPublishRequest>(TTarget instance, TPublishRequest request)
+        {
+            if (request is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var requestProxy = request.DuckCast<IPublishRequest>();
+
+            var scope = AwsSnsCommon.CreateScope(Tracer.Instance, Operation, SpanKinds.Producer, out AwsSnsTags tags);
+            tags.TopicArn = requestProxy.TopicArn;
+            tags.TopicName = AwsSnsCommon.GetTopicName(requestProxy.TopicArn);
+
+            if (scope?.Span.Context != null)
+            {
+                ContextPropagation.InjectHeadersIntoMessage<TPublishRequest>(requestProxy, scope.Span.Context);
```

Comment:
```suggestion
                ContextPropagation.InjectHeadersIntoMessage(requestProxy, scope.Span.Context);
```

Reply:
is this just a nit or does it serve some perf benefit? curious as im still learning dotnet

Reply:
@zARODz11z I'd suggest reading about this in https://github.com/DataDog/dd-trace-dotnet/blob/master/docs/development/DuckTyping.md to learn about the differences

Nonetheless, after a chat with Andrew Lock, we've agreed that the suggested changes to change boxing into using generic type constraints are not needed, since 
> you can only do that if the types you're passing in aren't being created using `DuckCast`.

Reply:
Resolving due to [this comment](https://github.com/DataDog/dd-trace-dotnet/pull/4712#issuecomment-1762212524).